packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -126,32 +126,52 @@ async function ssrTransformScript(
   ) {
     const source = importNode.source.value as string
     deps.add(source)
-    const importId = `__vite_ssr_import_${uid++}__`
 
     // Reduce metadata to undefined if it's all default values
-    if (
-      metadata &&
-      (metadata.importedNames == null || metadata.importedNames.length === 0)
-    ) {
-      metadata = undefined
-    }
-    const metadataStr = metadata ? `, ${JSON.stringify(metadata)}` : ''
-
-    s.update(
-      importNode.start,
-      importNode.end,
-      `const ${importId} = await ${ssrImportKey}(${JSON.stringify(
-        source,
-      )}${metadataStr});\n`,
-    )
+    const metadataArg =
+      (metadata?.importedNames?.length ?? 0) > 0
+        ? `, ${JSON.stringify(metadata)}`
+        : ''
 
-    if (importNode.start === index) {
-      // no need to hoist, but update hoistIndex to keep the order
-      hoistIndex = importNode.end
-    } else {
-      // There will be an error if the module is called before it is imported,
-      // so the module import statement is hoisted to the top
+    const importId = `__vite_ssr_import_${uid++}__`
+    const transformedImport = `const ${importId} = await ${ssrImportKey}(${JSON.stringify(
+      source,
+    )}${metadataArg});`
+
+    s.update(importNode.start, importNode.end, transformedImport)
+
+    // Check for non-whitespace characters between the last import and the
+    // current one, to determine if hoisting is needed.
+    // TODO: Account for comments between imports.
```

Comment:
```suggestion
    // If there's only whitespace characters between the last import and the
    // current one, that means there's no statements between them and
    // hoisting is not needed.
    // FIXME: account for comments between imports
```
We can say it in the other way ("we assume the non-whitespace character is a statement"), but I guess it's more easier to understand in this way.